internal/provider/resource/integration_gcp_secret_manager.go

```diff
@@ -0,0 +1,380 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	infisical "terraform-provider-infisical/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &IntegrationGcpSecretManagerResource{}
+)
+
+// NewProjectResource is a helper function to simplify the provider implementation.
+func NewIntegrationResource() resource.Resource {
```

Comment:
we might need to add `Gcp` prefix here so that this resource is unique within the package. there will be conflict when we add more integrations